### ISSUE TITLE
Fatal divide by zero error when quantity is 0 in PHP8 

### ISSUE THREAD

**Issue by dmallory42** [Automattic/woocommerce-subscriptions-core#197]
## Describe the bug

In #177, a fix was implemented to prevent a divide by zero error if the store is running on PHP8. The following code is used to check for this:

```php
// In rare cases quantity can be zero. Check first to prevent triggering a fatal error in php8+
                if ( 0 !== $item_to_renew['qty'] ) {
                    $_product->set_price( $price / $item_to_renew['qty'] );
                }
```

However, it looks like `$item_to_renew['qty']` is not an integer (presumably it is a string), so this check doesn't have the desired effect. We had a customer report (5461670-zd-woothemes) reporting a fatal error and proposing the following fix:

```php
// In rare cases quantity can be zero. Check first to prevent triggering a fatal error in php8+
                if ( 0 !== intval($item_to_renew['qty']) ) {
                    $_product->set_price( $price / intval($item_to_renew['qty']) );
                }
```

This should prevent this error happening in all cases.

## To Reproduce

See #175 (alternatively, the customer who raised this issue suggested it occurs when a new customer logs into their store).

### Expected behavior

We don't see any fatal errors related to division by zero errors.

- [x] Does this issue affect WooCommerce Subscriptions? yes: 5461670-zd-woothemes
- [ ] Does this issue affect WooCommerce Payments? TBC

**Comment by darcie** [comment]
36403807-hc is seeing this, moved to 5466737-zen to get followups. 

Error comes when changing payment method in the customer's account. 

```
2022-08-09T04:50:53+00:00 CRITICAL Uncaught DivisionByZeroError: Division by zero in /usr/www/users/nnnn/wp-content/plugins/woocommerce-subscriptions/vendor/woocommerce/subscriptions-core/includes/class-wcs-cart-renewal.php:458
Stack trace:
#0 /usr/www/users/nnnn/wp-includes/class-wp-hook.php(307): WCS_Cart_Renewal->get_cart_item_from_session(Array, Array, '1dccb3392487593...')
#1 /usr/www/users/nnnn/wp-includes/plugin.php(191): WP_Hook->apply_filters(Array, Array)
#2 /usr/www/users/nnnn/wp-content/plugins/woocommerce/includes/class-wc-cart-session.php(163): apply_filters('woocommerce_get...', Array, Array, '1dccb3392487593...')
#3 /usr/www/users/nnnn/wp-includes/class-wp-hook.php(307): WC_Cart_Session->get_cart_from_session('')
#4 /usr/www/users/nnnn/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters(NULL, Array)
#5 /usr/www/users/nnnn/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#6 /usr/www/users/nnnn/wp-settings.php(620): do_action('wp_loaded')
#7 /usr/www/users/nnnn/wp-config.php(96): require_once('/usr/www/users/...')
#8 /usr/www/users/nnnn/wp-load.php(50): require_once('/usr/www/users/...')
#9 /usr/www/users/nnnn/wp-blog-header.php(13): require_once('/usr/www/users/...')
#10 /usr/www/users/nnnn/index.php(17): require('/usr/www/users/...')
#11 {main}
thrown in /usr/www/users/nnnn/wp-content/plugins/woocommerce-subscriptions/vendor/woocommerce/subscriptions-core/includes/class-wcs-cart-renewal.php on line 458
```

**Comment by kaushikasomaiya** [comment]
5501207-zen

CC: @brucealdridge @haszari

**Comment by scottbuscemi** [comment]
+1'ing this issue - causes many fatal errors for folks renewing, logging in, or just trying to view their accounts.

**Comment by haszari** [comment]
Marking as critical since there have been multiple reports of a fatal error.

@dmallory42 @brucealdridge @scottbuscemi  @kaushikasomaiya @darcie Can anyone reproduce this at will? 

We'll need to be able to reproduce to confirm any fix before release. 

- **What are the flows / steps for each of the tickets?** 
- **What plugins are installed + active?** (WC Subscriptions or WCPay Subscriptions or both)

Please add that info to this bug, and all relevant plugins. #175 is very complex (unusual, edge case) flow so I suspect merchants are affected in a different way.

Next steps:

- Test on Pressable, Atomic or Jurassic Ninja with PHP8
- Determine reproduce steps.

Then we can work on a fix and get a patch release out as soon as possible.

**Comment by kaushikasomaiya** [comment]
Hi @haszari @brucealdridge 

Like I mentioned in my original issue: #[4359](https://github.com/woocommerce/woocommerce-subscriptions/issues/4359), this is likely due to cart sessions getting mixed up.

To reproduce

- Buy a Subs product with any quantity
- Buy same product or a different one with any quantity
- Go to WP Admin - and create a pending renewal order for both the subscriptions
- In a private window, login with the same account and act as a customer
- Go to My Account > Subscriptions and open both the subscriptions side by side
- Press Pay button on both the browser tabs to go to cart
- Pay for one and close the tab
- Try paying for the other one, you'll get a fatal error and site crashes 
- Further, you will not be able to access frontend site unless customer cart sessions are cleared up from Tools.

I am not sure if there are other ways users might be getting affected. I can research a bit and get back.

 I hope this helps! :)

**Comment by brucealdridge** [comment]
This has now been released in [WooCommerce Subscriptions 4.5.1](https://woocommerce.com/products/woocommerce-subscriptions/)
